mlflow/store/file_store.py

```diff
@@ -407,12 +407,19 @@ def _get_run_files(self, run_uuid, resource_type):
     @staticmethod
     def _get_metric_from_file(parent_path, metric_name):
         _validate_metric_name(metric_name)
-        metric_data = read_file_lines(parent_path, metric_name)
+        metric_data = [
+           line.strip().split(" ") for line in read_file_lines(parent_path, metric_name)
+        ]
         if len(metric_data) == 0:
-            raise Exception("Metric '%s' is malformed. No data found." % metric_name)
-        last_line = metric_data[-1]
-        timestamp, val = last_line.strip().split(" ")
-        return Metric(metric_name, float(val), int(timestamp))
+            raise ValueError("Metric '%s' is malformed. No data found." % metric_name)
+        max_timestamp, max_value = metric_data[0]
+        for timestamp, value in metric_data[1:]:
+            if timestamp > max_timestamp:
```

Comment:
It looks like you're comparing the timestamps and values before converting them into numeric types, which probably won't in general give the right answer. You probably need to first convert them into numbers above with something like:

```python
metric_data = []
for line in read_file_lines(parent_path, metric_name):
    timestamp_string, value_string = line.split()
    metric_data.append(
        (int(timestamp_string), float(value_string))
    )
```

and then do the ordering.

You might also want to consider simplifying the logic by using the fact that the Python builtin `sorted` will sort tuples by their first entry, then by their second, etc.:

```python
>>> tuples = [(2, 0), (1, 1), (0, 1), (0, 0), (1, 0), (2, 1)]
>>> sorted(tuples)
[(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1)]
```

this means that your logic could be simplified to:

```python
max_timestamp, max_value = sorted(metrics_data)[-1]
```

(with the conversion of strings to numbers as I described above).